gnocchi/incoming/file.py

```diff
@@ -35,6 +35,11 @@ def __init__(self, conf, greedy=True):
         self.basepath = conf.file_basepath
         self.basepath_tmp = os.path.join(self.basepath, 'tmp')
 
+        # We need to make sure that the 'tmp' path exists; otherwise,
+        # we will have an issue when running the method "set_storage_settings"
+        # that created the temporary JSON file under the 'tmp' folder.
+        utils.ensure_paths([self.basepath_tmp])
```

Comment:
this seems to already exists in the upgrade method? i guess we need to have it run before the super() call.
i'd lean more to fixing it there so we don't create folders whenever we create instance.

Reply:
I see. Done.

Thanks for the review!